src/Player.Vm.Api/Features/Callbacks/CallbackController.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Carnegie Mellon University. All Rights Reserved.
+// Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Swashbuckle.AspNetCore.Annotations;
+using Player.Vm.Api.Domain.Vsphere.Models;
+using MediatR;
+using Player.Vm.Api.Domain.Models;
+using Player.Vm.Api.Features.Vms;
+using Player.Api.Client;
+using Player.Vm.Api.Infrastructure.BackgroundServices;
+using Newtonsoft.Json;
+
+namespace Player.Vm.Api.Features.Callbacks
+{
+    public class CallbacksController : ControllerBase
+    {
+        private readonly IVmService _vmService;
+        private readonly ICallbackBackgroundService _backgroundService;
+
+        public CallbacksController(IVmService vmService, ICallbackBackgroundService backgroundService)
+        {
+            _vmService = vmService;
+            _backgroundService = backgroundService;
+        }
+
+        /// <summary>
+        /// Receive a webhook event
+        /// </summary>
+        /// <param name="evt"></param>
+        /// <param name="ct"></param>
+        [HttpPost("api/callback")]
+        [ProducesResponseType((int) HttpStatusCode.Accepted)]
+        [SwaggerOperation(OperationId = "respond")]
+        public IActionResult Respond([FromBody] WebhookEvent evt, CancellationToken ct)
+        {
+            _backgroundService.AddEvent(evt);
```

Comment:
I think we should throw the message into the db before returning Accepted and then have the background task remove it from the db when it finishes so that we don't have a problem if this app gets restarted or crashes after having told the caller that it accepted the message but before it finished processing it.